lib/galaxy/managers/workflows.py

```diff
@@ -924,6 +925,15 @@ def __connect_workflow_steps( self, steps, steps_by_external_id ):
 
             del step.temp_input_connections
 
+    def __set_default_label( self, module, state ):
+        """ Previously data input modules had a `name` attribute to rename individual steps. Here, this value is transferred
+        to the actual `label` attribute which is available for all module types, unique, and mapped to its own database column.
+        """
+        if not module.label and module.type in [ 'data_input', 'data_collection_input' ] and isinstance( state, dict ):
+            default_label = state.get( 'name' )
+            if str( default_label ).lower() != module.name.lower():
+                module.label = default_label
```

Comment:
Yes, it would and its much better to do that, otherwise it might cause issues if someone decides to change the fixed module name. Thanks a lot.